packages/tools/kadena-cli/src/utils/createOption.ts

```diff
@@ -1,56 +1,91 @@
+ *
+ * @param previousQuestions - The answers or data collected from previous prompts.
+ * @param args - Command line arguments or other parameters influencing the prompt's behavior.
+ * @param isOptional - Indicates whether the prompt is optional.
+ * @returns A promise resolving to the user's input or a specific value indicating a skip or creation action.
+ */
+export type IPrompt<T> = (
+  previousQuestions: Record<string, unknown>,
+  args: Record<string, unknown>,
+  isOptional: boolean,
+) => T | Promise<T>;
 
 export interface IOptionCreatorObject {
-  prompt: IPrompt;
-  // eslint-disable-next-line @typescript-eslint/no-explicit-any
-  validation: any;
+  key: string;
+  prompt: IPrompt<unknown>;
+  validation: z.ZodSchema;
   option: Option;
-  // eslint-disable-next-line @typescript-eslint/no-explicit-any
-  expand?: (label: string) => any;
-  // eslint-disable-next-line @typescript-eslint/no-explicit-any
-  transform?: (key: string) => any;
+  expand?: (label: string) => unknown;
+  transform?: (key: string) => unknown;
+  defaultIsOptional?: boolean;
 }
 
 // eslint-disable-next-line @typescript-eslint/explicit-function-return-type
-export function createOption<T extends IOptionCreatorObject>(option: T) {
+export function createOption<const T extends IOptionCreatorObject>(data: T) {
   return (
     {
-      isOptional = true,
+      isOptional,
       disableQuestion: isInQuestions = true,
     }: {
       isOptional?: boolean;
       disableQuestion?: boolean;
-    } = { isOptional: true, disableQuestion: true },
+    } = { disableQuestion: true },
   ) => {
+    const {
+      key,
+      option,
+      expand,
+      transform,
+      prompt,
+      validation,
+      defaultIsOptional,
+    } = data;
+    const optional = isOptional ?? defaultIsOptional ?? true;
     return {
-      ...option,
-      isOptional,
-      isInQuestions,
+      key: key as T['key'],
+      option,
       // eslint-disable-next-line @typescript-eslint/no-explicit-any
-      prompt: (responses: any, args: any) =>
-        option.prompt(responses, args, isOptional),
-      validation: isOptional ? option.validation.optional() : option.validation,
+      expand: expand as T['expand'] extends (...args: any[]) => unknown
+        ? (value: Awaited<ReturnType<T['prompt']>>) => ReturnType<T['expand']>
+        : undefined,
+      // eslint-disable-next-line @typescript-eslint/no-explicit-any
+      transform: transform as T['transform'] extends (...args: any[]) => unknown
+        ? (
+            value: Awaited<ReturnType<T['prompt']>>,
+          ) => ReturnType<T['transform']>
+        : undefined,
+      isOptional: optional,
+      isInQuestions,
+      prompt: ((responses, args) =>
+        prompt(responses, args, optional)) as T['prompt'],
+      validation: (isOptional === true
+        ? validation.optional()
+        : validation) as T['validation'],
     };
   };
 }
+
+const testoption = createOption({
```

Comment:
`testOption` is test code, I will remove it after discussion about `createOption` is done.